src/kraken/core/system/context.py

```diff
@@ -231,26 +241,51 @@ def resolve_tasks(
             relative_to = relative_to.address if relative_to is not None else Address.ROOT
 
         if not relative_to.is_absolute():
-            raise ValueError(f"'relative_to' must be an absolute address (got {relative_to!r})")
+            raise TaskResolutionException(f"'relative_to' must be an absolute address (got {relative_to!r})")
```

Comment:
I would consider this a contract in the API, thus a non absolute Address is an invalid value to pass into the function.

It is something the caller can foresee; whereas they cannot foresee whether a valid address will resolve successfully or not.

```suggestion
            raise ValueError(f"'relative_to' must be an absolute address (got {relative_to!r})")
```